dvc/io/__init__.py

```diff
@@ -0,0 +1,152 @@
+import sys
+from collections import defaultdict
+from typing import Any, Dict, Iterable, Optional, TextIO
+
+from funcy import cached_property
+
+from dvc.progress import Tqdm
+from dvc.utils import colorize
+
+NEWLINE = "\n"
+SEP = " "
+
+
+class Formatter:
+    def __init__(self, theme: Dict = None, defaults: Dict = None) -> None:
+        theme = theme or {
+            "success": {"color": "green", "style": "bold"},
+            "warn": {"color": "yellow"},
+            "error": {"color": "red", "style": "bold"},
+        }
+        self.theme = defaultdict(lambda: defaults or {}, theme)
+
+    def format(self, message: str, style: str = None, **kwargs) -> str:
+        return colorize(message, **self.theme[style])
+
+
+class IO:
+    def __init__(
+        self,
+        formatter: Formatter = None,
+        output: TextIO = None,
+        error: TextIO = None,
+        disable: bool = True,
+    ) -> None:
+        self._input: TextIO = sys.stdin
+        self._output: TextIO = output or sys.stdout
+        self._error: TextIO = error or sys.stderr
+
+        self.formatter: Formatter = formatter or Formatter()
+        self.disabled: bool = disable
+
+    def success(self, message: str) -> None:
+        self.write(message, style="success")
+
+    def error(self, message: str) -> None:
+        self.error_write(message, style="error")
+
+    def warn(self, message: str) -> None:
+        self.error_write(message, style="warn")
```

Comment:
Is the idea here that we will eventually replace the use of `logging` with these methods?

Or are these just for console UI output messages which would be separate from internal logging calls?

Reply:
If we start printing in colors, they will come in handy.